src/main/java/cubicchunks/worldgen/ICubeGenerator.java

```diff
@@ -0,0 +1,142 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+
+package cubicchunks.worldgen;
+
+import cubicchunks.util.CubeCoords;
+import cubicchunks.world.column.Column;
+import cubicchunks.world.cube.Cube;
+import cubicchunks.worldgen.dependency.DependentCubeManager;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public interface ICubeGenerator {
+
+	/**
+	 * Returns the number of cubes that are queued for generation.
+	 *
+	 * @return The number of cubes that are queued for generation.
+	 */
+	int getQueuedCubeCount();
+
+	/**
+	 * Returns the GeneratorPipeline used by this ICubeGenerator. This pipeline dictates the stages every cube has to
+	 * go through and the processors used in each given stage. Modifications on this pipeline are strongly discouraged
+	 * as they are likely to disrupt generation of currently queued cubes.
+	 *
+	 * @return The GeneratorPipeline used by this ICubeGenerator.
+	 */
+	@Nonnull
+	GeneratorPipeline getGeneratorPipeline();
```

Comment:
As I said in previous comment, if GeneratorPipeline is now StageRegistry, it makes more sense if it's part of World.